contributors/design-proposals/cli/kubectl-unix-socket.md

```diff
@@ -0,0 +1,42 @@
+# Unix Socket Support for kubectl
+
+Status: Pending
+
+Version: GA
+
+Implementation Owner: @choo-stripe
+
+## Motivation
+
+Currently kubectl does not expose a way to send the request over a unix socket. Curl provides a --unix-socket option, as do many other cli utilities. Stripe uses unix sockets as a secured proxy into some infrastructure and it would be great to be able to send reqs from kubectl over this unix socket.
+
+## Proposal
+
+
+A --unix-socket flag will be added to kubectl, much like in curl. The curl docs state:
+(HTTP) Connect through this Unix domain socket, instead of using the network.
+
+This is precisely what the kubectl unix-socket flag will be used for.
+
+## User Experience
```

Comment:
I suggest we focus on the kubeconfig experience.

Reply:
Added an example - also tested on my machine